src/components/navbar/NavbarItemStyles.tsx

```diff
@@ -0,0 +1,49 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { css } from '@emotion/react';
+import styled from '@emotion/styled';
+
+export const StyledWrapper = styled.div`
+  padding: var(--echoes-dimension-space-150) 0
+    calc(var(--echoes-dimension-space-150) - var(--echoes-focus-border-width-default));
+
+  border-bottom: var(--echoes-focus-border-width-default) solid transparent;
+`;
+
+export const navbarItemStyle = css`
+  --hover: var(--echoes-color-text-default);
+
+  display: block;
+
+  padding: var(--echoes-dimension-space-75);
+  border-radius: var(--echoes-border-radius-200);
+
+  font: var(--echoes-typography-text-default-semi-bold);
+  text-decoration-line: var(--echoes-text-decoration-none);
+
+  &:hover {
+    background-color: var(--echoes-color-background-default-hover);
+  }
+
+  &:active {
+    background-color: var(--echoes-color-background-default-active);
+  }
+`;
```

Comment:
[non-blocking] I wonder if it would be better to make this a styled component, e.g.

```tsx
export const NavbarItemBase = styled.div`
  --hover: var(--echoes-color-text-default);
  display: block;
  ...
`;
```

And then use it like this

```tsx
<NavbarItemBase as={Button} {...rest}>
  {children}
</NavbarItemBase>
```

Or is TypeScript not able to infer props in that case?

Reply:
The ref would become an issue, I believe